cluster/src/main/java/org/apache/iotdb/cluster/server/member/DataGroupMember.java

```diff
@@ -816,6 +828,22 @@ private boolean createTimeseriesForFailedInsertion(InsertPlan plan)
       }
     }
 
+    if (plan instanceof InsertRowsPlan) {
+      for (InsertRowPlan insertPlan : ((InsertRowsPlan) plan).getInsertRowPlanList()) {
+        if (insertPlan.getFailedMeasurements() != null) {
+          insertPlan.getPlanFromFailed();
+        }
+      }
+    }
+
+    if (plan instanceof InsertRowsOfOneDevicePlan) {
+      for (InsertRowPlan insertPlan : ((InsertRowsOfOneDevicePlan) plan).getRowPlans()) {
+        if (insertPlan.getFailedMeasurements() != null) {
+          insertPlan.getPlanFromFailed();
```

Comment:
What's the difference for `getPlanFromFailed` and `recoverFromFailure`? I remember we used `recoverFromFailure` in `Coordinator` before.

Reply:
The method getPlanFromFailed can only get failed measurements without successful ones, so fewer time series will be processed afterwards.

Reply:
Can you be sure to return the correct error message(the correct index for a column) when a column is recovered with getPlanFromFailed and the execution fails again?